subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/transform/ArtifactTransformCachingIntegrationTest.groovy

```diff
@@ -1346,6 +1353,88 @@ allprojects {
         outputDir("snapshot-1.2-SNAPSHOT.jar", "snapshot-1.2-SNAPSHOT.jar.txt") != outputDir2
     }
 
+    def "cleans up cache"() {
+        given:
+        buildFile << """
+            allprojects {
+                dependencies {
+                    registerTransform {
+                        from.attribute(artifactType, "jar")
+                        to.attribute(artifactType, "size")
+                        artifactTransform(FileSizer)
+                    }
+                }
+                task resolve {
+                    def artifacts = configurations.compile.incoming.artifactView {
+                        attributes { it.attribute(artifactType, 'size') }
+                    }.artifacts
+                    inputs.files artifacts.artifactFiles
+                    doLast {
+                        println "files: " + artifacts.artifactFiles.collect { it.name }
+                        println "ids: " + artifacts.collect { it.id.displayName }
+                        println "components: " + artifacts.collect { it.id.componentIdentifier }
+                    }
+                }
+            }
+
+            class FileSizer extends ArtifactTransform {
```

Comment:
⭕️ Is there something we could reuse instead of having this test introduce its own transform and checking all the files, ids and components (which are not the topic of this test)?